src/compaction.c

```diff
@@ -98,34 +98,38 @@ void AvgReadContext(void *contextPtr, RedisModuleIO * io){
 void *StdCreateContext() {
     StdContext *context = (StdContext *)malloc(sizeof(StdContext));
     context->cnt = 0;
-    context->mean =0;
-    context->std =0;
+    context->sum = 0;
+    context->sum_2 = 0;
     return context;
 }
 
 void StdAddValue(void *contextPtr, double value){
     StdContext *context = (StdContext *)contextPtr;
-    double oldMean = context->mean;
-    context->mean = oldMean + (value - oldMean) / ++context->cnt;
-    context->std = context->std + (value - oldMean) * (value - context->mean);
+    ++context->cnt;
+    context->sum += value;
+    context->sum_2 += value * value; 
 }
 
 double VarPopulationFinalize(void *contextPtr) {
     StdContext *context = (StdContext *)contextPtr;
-    double res = 0;
-    if(context->cnt > 0) {
-        res = context->std / context->cnt;
+    double sum = context->sum;
+    double sum_2 = context->sum_2;
+    uint64_t count = context->cnt;
+    if(count <= 1) {
+        return 0;
     }
-    return res;
+    return (sum_2 - 2 * (sum / count) * sum + pow(sum / count, 2) * count) / count;
```

Comment:
You can add here a comment something like:
`var(X) = sum((x_i - E[X])^2) =  sum(x_i)^2 -2sum(x_i)*E[X] + E^2[X]`

Reply:
Done!